### PR TITLE
Module

### PR DESCRIPTION
**Pull request by Farnazmdi**

### CODE REVIEW

**Comment by Farnazmdi** [comment]
@aarmey I ran the gene module (twice!) once with z-scoring the data and once without, both times the heatmap showed up like this: I don't know what I am doing wrong... Also, there are 1000 new files, I don't think I should commit them, I will save them on my computer. 
image

File: tfac/figures/figure3.py [hunk 1]
```diff
@@ -17,3 +19,14 @@ def clustergram_proteins():
     decreased_proteins = proteins.loc[((-0.001>=proteins).any(1) | (proteins>= 0.001).any(1))]
     g = sns.clustermap(decreased_proteins, cmap="bwr", method="centroid", center=0, figsize=(14, 20))
     plt.savefig("output/clustergram_proteins.svg")
+
+def gene_module_enrichm_plot():
+    """ Plots the gene module analysis in heatmap, in the tfac/output folder as modules_v_components.svg """
+    ns, data = ns_RNAseq_data()
+    # z-scoring each column of the data
+    for col in data.columns[1:]:
+        data[col] = (data[col] - np.mean(data[col])) / np.std(data[col])
```
Comment (aarmey) [inline tfac/figures/figure3.py:28]: Do you mean you're z-scoring the genes before module analysis, or the modules after the analysis? It looks like you're doing it after the analysis here...

File: tfac/dataHelpers.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def importLINCSprotein():
 
 def ohsu_data():
     """ Import OHSU data for PARAFAC2"""
-    return pd.read_csv(join(path_here, "tfac/data/ohsu/MDD_RNAseq_Level4.csv"))
+    return pd.read_csv(join(path_here, "tfac/data/ohsu/RNAseq_Level4.txt"), delimiter="\t")
```
Comment (aarmey) [inline tfac/dataHelpers.py:24]: What is the difference between these files?

**Review by aarmey — CHANGES_REQUESTED** [review]

File: tfac/dataHelpers.py (hunk 2 above)
Comment (Farnazmdi) [reply tfac/dataHelpers.py:24]: The WGCNA needs the data to be tab-delimited. I saved the data in a tab-delimited format and renamed it

File: tfac/figures/figure3.py (hunk 1 above)
Comment (Farnazmdi) [reply tfac/figures/figure3.py:28]: I did it before.. I divided the code into few functions, the `ns_RNAseq_data()` returns the imported data and the namespaces that `run_module()` will use. I z-scored before passing to the run_module().. is it wrong?

**Comment by Farnazmdi** [comment]
Also regarding the build and test failure, I tried to update the package versions based on master, since there are lots of packages needed for this PR, but the dependencies seem to have conflicts.

**Comment by aarmey** [comment]
What are the new packages the code directly uses? Those are the only ones you should add to requirements. Dependencies will automatically be installed.

File: tfac/figures/figure3.py (hunk 1 above)
Comment (aarmey) [reply tfac/figures/figure3.py:28]: We'll you definitely don't want to z-score after running the module analysis. I agree the results here don't look right. Maybe check with @JacksonLChin about what he did?

File: tfac/dataHelpers.py (hunk 2 above)
Comment (aarmey) [reply tfac/dataHelpers.py:24]: Aha, ok.

**Comment by Farnazmdi** [comment]
I ran the module again and it is working I think.
image
I will save the files on my computer. Is this ready to merge?

**Comment by aarmey** [comment]
Great. What changed?

**Comment by Farnazmdi** [comment]
I had to "save" the z-scored data.. since the WGCNA accesses the data on its own. It is exactly what you had said.

**Comment by Farnazmdi** [comment]
Could I merge this?

**Review by aarmey — APPROVED** [review]
Sure.

**Pull request merged**